lib/openhab/dsl/rules/terse.rb

```diff
@@ -57,6 +56,42 @@ def #{trigger}(*args, name: nil, id: nil, **kwargs, &block)     # def changed(*a
         def_terse_rule(:thing_removed)
         def_terse_rule(:updated)
         def_terse_rule(:on_start)
+
+        module_function
+
+        #
+        # Create a rule that keeps the item up to date according to the given block
+        #
+        # @see Core::EntityLookup.capture_items See capture_items for caveats of how to access items in your block.
+        #
+        # @param [Item] item The item to keep up to date
+        # @param [String, nil] name The name for the rule
+        # @param [String, nil] id The id for the rule
+        # @yield Calculate the new value for the item
+        # @yieldreturn [State]
+        # @return [Core::Rules::Rule]
+        #
+        # @example
+        #  calculated_item(Furnace_DeltaTemp) { FurnaceSupplyAir_Temp.state - FurnaceReturnAir_Temp.state }
+        #
+        def calculated_item(item, name: nil, id: nil, &block)
+          raise ArgumentError, "Block is required" unless block
+
+          items = Core::EntityLookup.capture_items do
+            item.ensure.update(yield)
+          end
+
+          name ||= "Calculate the value of #{item}"
+          id ||= NameInference.infer_rule_id_from_block(block)
+          script = block.source rescue nil # rubocop:disable Style/RescueModifier
+
+          rule name, id: id, script: script, binding: block.binding do
+            changed(*items)
+            run do
+              item.ensure.update(yield)
```

Comment:
the problem is it’s not so easy. I have to run the block essentially as an on_load in order to identify the items. but I can’t check their state is not nil until I know what items I’m looking at

I can’t even catch NoMethodError, because that will interrupt execution of the block, so I won’t know what other items were accessed

I could, only within the context of a capture_items block, instead of returning nil return a special object that delegates to nil, but swallows any NoMethodErrors. but that sounds super error-prone

I think a better option would be to allow explicitly specifying the dependencies, and then if you do that, have an option for require_state: true

WDYT?

Reply:
I'm going to noodle on this a bit. even if you make the user have their block be written to be safe in the face of `UnDefType`, it would likely return `nil`, and it might be the case that when that happens they want to _not_ update the item, instead of updating it to `NULL`. but automatic item capture is like 90% of the appeal of this feature

Reply:
waiting for the result of your noodling